segmenters/image/FaceNetSegmenter/__init__.py

```diff
@@ -76,27 +73,23 @@ def segment(self, data: 'np.ndarray', *args, **kwargs) -> List[Dict]:
         """
         if self.channel_axis != self._default_channel_axis:
             data = np.moveaxis(data, self.channel_axis, self._default_channel_axis+1)
-        data = np.asarray(data)
-        with torch.no_grad():
-
-            image = torch.from_numpy(data.astype('float32')).to(self.device)
-
 
-            faces, probabilities = self.face_detector(image, return_prob=True)
-
-            print(f"probabilities : {probabilities[0]}")
-            if self.keep_all:
-                # All faces and probabilities are grouped in the first dimension of the first batch element
-                faces = faces[0]
-                probabilities = probabilities[0]
-
-            results = [
-                dict(
-                    offset=0,
-                    weight=probability,
-                    blob=face.detach().numpy(),
-                )
-                for face, probability in zip(faces, probabilities)
-                if face is not None
-            ]
-            return results
+        batch = np.copy(data)
+        results = []
+        with torch.no_grad():
+            image_batch = torch.from_numpy(batch.astype('float32')).to(self.device)
+            facesBatch, probabilitiesBatch = self.face_detector(image_batch, return_prob=True)
+            for faces, probabilities in zip(facesBatch, probabilitiesBatch):
+                if faces is not None:
+                    batched = []
+                    for face, probability in zip(faces, probabilities):
+                        if face is not None:
```

Comment:
and here?

Reply:
this is most likely not possible. If they have returned a list for `faces` it will not contain None

Reply:
I see this is not needed.